### PR TITLE
GreatestCommonFactor: Don't trigger on single numbers, Fix #3407

### LINKED ISSUE #3407: Random Date: Allow some customization
For example, it'd be good to be able to restrict the date to certain years or time periods, e.g.:
`random date (in) 2016`
`random date (in the) past`
`random date (in the) future`   

------
IA Page:  http://duck.co/ia/view/random_date 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @GuiltyDolphin

### PR DESCRIPTION
**Pull request by jmwoliver**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [ ] New Instant Answer
    - [ ] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [X] Improvement
    - [X] Bug fix: **`GreatestCommonFactor: Fix #3407, Don't trigger on single numbers`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

Don't trigger on single numbers

###### Which issues (if any) does this fix?

Fixes #3407 - I made it so that it outputs "Cannot compute Greatest common factor" if user only inputs one number. On the discussion, it said that it is not usually done that way, so I can change it if need be.

###### People to notify @zekiel @GuiltyDolphin @austinheimark @procommand

---

Instant Answer Page: https://duck.co/ia/view/greatest_common_factor

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html):  @austinheimark

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by jmwoliver** [comment]
The pull request is failing because I didn't update the test file. I will wait to do that until I am told if my current method is okay.

File: lib/DDG/Goodie/GreatestCommonFactor.pm [hunk 1]
```diff
@@ -18,23 +18,37 @@ handle remainder => sub {
...
-      structured_answer => {
-	data => {
-	  title => "$result",
-	  subtitle => "Greatest common factor: $formatted_numbers"
-	},
-	templates => {
-	  group => "text",
-	}
+    if (@numbers == 1) {
+      return "Cannot return Greatest Common Factor.",
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/GreatestCommonFactor.pm]: We shouldn't be returning a response. All that is requried is a one-liner to return if there is only one element.

As a general rule IAs ***should not*** return explanations as to why they are not returning a 'result'.

**Comment by jmwoliver** [comment]
@GuiltyDolphin is it just something like this?

**Comment by GuiltyDolphin** [comment]
@jmwoliver Just `return unless @numbers > 1`.

**Comment by jmwoliver** [comment]
@GuiltyDolphin Oh okay, thank you! What exactly do I need to put in the test cases?

**Comment by GuiltyDolphin** [comment]
@jmwoliver Well, we don't want it to *trigger* on the queries, in which case the result will be `undef`ined.

**Comment by jmwoliver** [comment]
@GuiltyDolphin Do I have it set up right now to where it will return undef? Also what do I put in the quotes? Would it be like this?

`'gcf 2'                           => build_test('', undef),`

**Comment by GuiltyDolphin** [comment]
@jmwoliver For queries that should not trigger we use `'query' => undef`; using Perl's `undef`ined value.

**Comment by jmwoliver** [comment]
@GuiltyDolphin Sorry for all the confusion, this is my first time with Perl. Should be good to go now! Do I need to be assigned to the bug page (https://github.com/duckduckgo/zeroclickinfo-goodies/issues/3047) or does that not matter? Thank you for all of your help!

**Comment by GuiltyDolphin** [comment]
@jmwoliver Thanks :dolphin:

**Comment by jmwoliver** [comment]
@Thank you! I appreciate all of your help!

**Pull request merged**